src/scratch/UCERF3/inversion/InversionFaultSystemRupSet.java

```diff
@@ -150,16 +154,50 @@ public InversionFaultSystemRupSet(LogicTreeBranch branch, SectionClusterList sec
 		this.sectionConnectionsListList = sectionClusterList.getSectionConnectionsListList();
 
 		// check that indices are same as sectionIDs (this is assumed here)
-		for(int i=0; i<faultSectionData.size();i++)
+		for (int i = 0; i < faultSectionData.size(); i++)
 			Preconditions.checkState(faultSectionData.get(i).getSectionId() == i,
 					"RupsInFaultSystemInversion: Error - indices of faultSectionData don't match IDs");
 
 		// calculate rupture magnitude and other attributes
 		calcRuptureAttributes(faultSectionData, sectionClusterList);
 	}
-	
+
+	/**
+	 * Helper method to create a InversionFaultSystemRupSet given a rupture set.
+	 * 
+	 * Notes: 
+	 * 1)this method is an interim solution until a proper constructor is
+	 * conceived. 
+	 * 2) test scratch.UCERF3.inversion.laughTest.TestIncrementalVsFullTests 
+	 * depends on OldPlausablityFilter.
+	 * 
+	 * @param {FaultSystemRupSet} rupSet
+	 * @param {LogicTreeBranch} branch
+	 * @return {InversionFaultSystemRupSet} 
+	 */
+	public static InversionFaultSystemRupSet fromRuptureSet(FaultSystemRupSet rupSet, LogicTreeBranch branch) {
```

Comment:
move to your NSHMP subclass?

Reply:
@voj @kevinmilner this has had major rework: 
 - removed the confusing reformatting
 - commented the single private->protected visibilty to indicate why I'm doing this. The over-ridden method [lives here](https://github.com/GNS-Science/nshm-nz-opensha/blob/ec0b1e578bdbc93d2d61b7adbafe64c8c007aa84/src/main/java/nz/cri/gns/NSHM/opensha/inversion/NSHM_InversionFaultSystemRuptSet.java#L46)
 - I've replaced the fromRuptureSet() method with a new constructor that uses **set/getClusterRuptures** as you suggested above. It also supports the case where **getClusterRuptures**() returns null, e.g. if no cluster_ruptures.json. 
 - there are [some integration tests](https://github.com/GNS-Science/nshm-nz-opensha/blob/14ef3bee4f45367f297c10953c6acc3aac018a30/src/integration/java/nz/cri/gns/NSHM/opensha/inversion/NSHMInversionRunnerIntegration.java) here which we could promote to this repo.


FYI I'm keen to know if this last one is OK. I'm trying to avoid the **InversionFaultSystemRupSetFactory** class for now as it's rather complex.